### PR TITLE
Initial stable version

### PR DESCRIPTION
**Pull request by leonardbf**
Initial stable (non-preview) version
Upgrade to R4 RP standard
Add availability functions

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5845 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5845/commits/beacb16057ead8f2c2ec62fd0389ef39317e53a8

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-05-01/netapp.json [hunk 1]
```diff
@@ -0,0 +1,2119 @@
...
+        {
+          "$ref": "#/parameters/PoolName"
+        },
+        {
+          "$ref": "#/parameters/ApiVersionParameter"
+        }
+      ],
+      "get": {
+        "summary": "Describe a Capacity Pool",
+        "description": "Get detailsof the specified capacity pool",
```
Comment (nschonni) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-05-01/netapp.json]:
```suggestion
        "description": "Get details of the specified capacity pool",
```

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-05-01/netapp.json [hunk 2]
```diff
@@ -0,0 +1,2119 @@
...
+	  "title": "Snapshot ID",
+	  "type": "string",
+	  "description": "UUID v4 used to identify the Snapshot",
+	  "pattern": "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$",
+	  "maxLength": 36,
+	  "minLength": 36,
+	  "example": "9760acf5-4638-11e7-9bdb-020073ca3333"
+        },
+	"baremetalTenantId": {
+	  "title": "Baremental Tenant ID",
```
Comment (nschonni) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-05-01/netapp.json]:
```suggestion
	  "title": "Baremetal Tenant ID",
```

Comment (leonardbf) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-05-01/netapp.json]: ok, will update.

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-05-01/netapp.json (hunk 1 above)
Comment (leonardbf) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-05-01/netapp.json]: ok, will update.

**Comment by leonardbf** [comment]
@nschonni I see that I get failures in the public.rest-api-specs check. These have not previously been present - is it a new check? In any case I've tried resolving them but with no success. Is there a way to run this locally as I've pushed to the repo each time to test. Currently I locally use an autorest validation on the json, and oav on the spec and examples but it doesn't cover this. Perhaps you could take a look at the issue in question too please. It seems there are missing required fields. I've tried adding them in various ways but nothing seems to work. I assume there is some basic fix I'm missing. Thanks.

**Comment by dsgouda** [comment]
Thanks @nschonni !

**Comment by dsgouda** [comment]
> 
> 
> @nschonni I see that I get failures in the public.rest-api-specs check. These have not previously been present - is it a new check? In any case I've tried resolving them but with no success. Is there a way to run this locally as I've pushed to the repo each time to test. Currently I locally use an autorest validation on the json, and oav on the spec and examples but it doesn't cover this. Perhaps you could take a look at the issue in question too please. It seems there are missing required fields. I've tried adding them in various ways but nothing seems to work. I assume there is some basic fix I'm missing. Thanks.

@leonardbf Let me take a look at that

**Comment by dsgouda** [comment]
@leonardbf The error is originating [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=19442) the examples will have to be corrected.
Instructions to use the model validation tool can be found [here](https://github.com/Azure/oav/blob/master/README.md)

**Comment by leonardbf** [comment]
> @leonardbf The error is originating [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=19442) the examples will have to be corrected.
> Instructions to use the model validation tool can be found [here](https://github.com/Azure/oav/blob/master/README.md)

@dsgouda thanks yes I had already seen where the failure was occurring and that it is due to required fields but I have been unable to apply the missing fields in a way that removes the failure. For instance the check complains about missing location but I did add that both in the parameters and nested in property. With reference to the tools, yes those are the ones that I run locally but they do not highlight the particular failure. I've ran the oav validate-example but it does not find these issues.

**Comment by dsgouda** [comment]
> 
> 
> > @leonardbf The error is originating [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=19442) the examples will have to be corrected.
> > Instructions to use the model validation tool can be found [here](https://github.com/Azure/oav/blob/master/README.md)
> 
> @dsgouda thanks yes I had already seen where the failure was occurring and that it is due to required fields but I have been unable to apply the missing fields in a way that removes the failure. For instance the check complains about missing location but I did add that both in the parameters and nested in property. With reference to the tools, yes those are the ones that I run locally but they do not highlight the particular failure. I've ran the oav validate-example but it does not find these issues.

From the looks of it, you have defined `location` to be a property on the `body` parameter but you have defined it as a parameter outside the `body` parameter in the examples. That should be the problem

**Comment by leonardbf** [comment]
> > > @leonardbf The error is originating [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=19442) the examples will have to be corrected.
> > > Instructions to use the model validation tool can be found [here](https://github.com/Azure/oav/blob/master/README.md)
> > 
> > 
> > @dsgouda thanks yes I had already seen where the failure was occurring and that it is due to required fields but I have been unable to apply the missing fields in a way that removes the failure. For instance the check complains about missing location but I did add that both in the parameters and nested in property. With reference to the tools, yes those are the ones that I run locally but they do not highlight the particular failure. I've ran the oav validate-example but it does not find these issues.
> 
> From the looks of it, you have defined `location` to be a property on the `body` parameter but you have defined it as a parameter outside the `body` parameter in the examples. That should be the problem

@dsgouda yes my first attempt to solve this was with location in the body but it still failed. The current state is the last attempt of some different combinations. Could you provide the exact command and syntax for running this failing check locally please.

**Comment by dsgouda** [comment]
> 
> 
> > > > @leonardbf The error is originating [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=19442) the examples will have to be corrected.
> > > > Instructions to use the model validation tool can be found [here](https://github.com/Azure/oav/blob/master/README.md)
> > > 
> > > 
> > > @dsgouda thanks yes I had already seen where the failure was occurring and that it is due to required fields but I have been unable to apply the missing fields in a way that removes the failure. For instance the check complains about missing location but I did add that both in the parameters and nested in property. With reference to the tools, yes those are the ones that I run locally but they do not highlight the particular failure. I've ran the oav validate-example but it does not find these issues.
> > 
> > 
> > From the looks of it, you have defined `location` to be a property on the `body` parameter but you have defined it as a parameter outside the `body` parameter in the examples. That should be the problem
> 
> @dsgouda yes my first attempt to solve this was with location in the body but it still failed. The current state is the last attempt of some different combinations. Could you provide the exact command and syntax for running this failing check locally please.

Did you run into the exact some error when you put the `location` property inside `body` parameter? 
@leonardbf Could you share what command you are running to locally validate the examples? 
@sarangan12 @veronicagg can someone help @leonardbf with using the model validator?

**Comment by leonardbf** [comment]
> > > > > @leonardbf The error is originating [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=19442) the examples will have to be corrected.
> > > > > Instructions to use the model validation tool can be found [here](https://github.com/Azure/oav/blob/master/README.md)
> > > > 
> > > > 
> > > > @dsgouda thanks yes I had already seen where the failure was occurring and that it is due to required fields but I have been unable to apply the missing fields in a way that removes the failure. For instance the check complains about missing location but I did add that both in the parameters and nested in property. With reference to the tools, yes those are the ones that I run locally but they do not highlight the particular failure. I've ran the oav validate-example but it does not find these issues.
> > > 
> > > 
> > > From the looks of it, you have defined `location` to be a property on the `body` parameter but you have defined it as a parameter outside the `body` parameter in the examples. That should be the problem
> > 
> > 
> > @dsgouda yes my first attempt to solve this was with location in the body but it still failed. The current state is the last attempt of some different combinations. Could you provide the exact command and syntax for running this failing check locally please.
> 
> Did you run into the exact some error when you put the `location` property inside `body` parameter?
> @leonardbf Could you share what command you are running to locally validate the examples?
> @sarangan12 @veronicagg can someone help @leonardbf with using the model validator?

@dsgouda yes I run three different commands before pushing to the repo
oav validate-spec netapp.json
oav validate-example netapp.json
autorest --azure-validator --input-file=netapp.json --openapi-type=arm
These are executed form the .\stable\2019-05-01 area.
Perhaps there is some additional command I should be using or an option on the validate-example. I see the failing checks, https://dev.azure.com/azure-sdk/public/_build/results?buildId=19442, are using some script.

**Comment by leonardbf** [comment]
@dsgouda ok the example validation now passes however there are new failures in the lint job which don't look like they are part of my change https://travis-ci.org/Azure/azure-rest-api-specs/jobs/521954258

**Comment by dsgouda** [comment]
@leonardbf noted, looks like we have some issues with travis right now, will restart the builds in a while and check, can confirm issues are not related to your change

**Comment by dsgouda** [comment]
@sarangan12 @veronicagg could you take a look at [this](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/521954258) error

**Comment by nschonni** [comment]
@dsgouda I think that was fixed in https://github.com/Azure/azure-rest-api-specs/commit/cba23e11a5b08df3f0af7783f89a979f5ddcb1ae by bumping the Travis worker image. You'd need to pull in the latest master to get that though

**Comment by dsgouda** [comment]
startbuild

**Comment by veronicagg** [comment]
@dsgouda there's a dotnet issue with linter diff, @sergey-shandar is updating CI to address this issue soon. In the meantime you can take a look at an equivalent job at https://dev.azure.com/azure-sdk/public/_build/results?buildId=19733 which is running linterDiff in azureDevOps. We'll get this cleaned up soon.

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks good for the most part, might need an ARM review
The linter is passing in ADO as @veronicagg pointed out

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5893 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5893/commits/929dccb464569a60447e53c98acb82b85ca670b3

**Comment by leonardbf** [comment]
@dsgouda Thanks for working on pushing this through. I'm not quite clear however on the current status. Will this PR be merged? Also, I see that a dotnet PR has been created (and previously a python PR). I'm wondering how this can be done without the swagger having been merged, i.e. finalized. I also need to perform the self-service generation of other SDKs (https://aka.ms/openapihub) but don't know if I can go ahead yet.

**Comment by dsgouda** [comment]
@leonardbf From the looks of it these end points create/modify Azure resources which is why I am waiting for someone from the ARM (Azure resource management) team to take a look at the changes. If by EOD there is no word, I will go ahead and merge this PR.
As for the generation of SDKs that can happen only when this gets merged.
@KrisBash can we expedite this?

**Comment by leonardbf** [comment]
@dsgouda hi, what's the latest? Can this be merged?

**Comment by nschonni** [comment]
@leonardbf the typos still haven't been fixed

**Comment by leonardbf** [comment]
> @leonardbf the typos still haven't been fixed

@nschonni The two I saw "baremental" and "detailsof" were fixed. Have I missed some others? Or do I need to set the two discussion points to resolved? Thanks.

**Comment by nschonni** [comment]
@leonardbf my mistake, I must have missed the "Outdated" comment. I think there is a button to "Resolve" feedback if it is done

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by dsgouda** [comment]
@leonardbf Don't have the merge perms here, will get this checked in today

**Pull request merged**